pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -158,10 +157,7 @@ func (pl *InterPodAffinity) getExistingAntiAffinityCounts(ctx context.Context, p
 	processNode := func(i int) {
 		nodeInfo := nodes[i]
 		node := nodeInfo.Node()
-		if node == nil {
```

Comment:
To ensure it's safe enough, I'd suggest craft a racing test by add/delete-ing nodes and meanwhile scheduling a number of pods.

IIRC we hit some nil Node issue before. @alculquicondor @ahg-g did you ring a bell?

Reply:
Thank you for your prompt reply

yes, I will add this test in the next few days

Reply:
We haven't faced these issues in a while, but a test would be useful.

Reply:
@Huang-Wei @alculquicondor 
the test code has been submitted, PTAL, thanks.

Sorry for the delay. I'm not sure if I have expressed the testing scenario you described. Please let me know if there are any improvements.